org.adempiere.base/src/org/compiere/model/MRole.java

```diff
@@ -115,16 +115,15 @@ else if (defaultRole.getAD_Role_ID() != AD_Role_ID
 	 * @param defaultRole
 	 */
 	private static void setDefaultRole(MRole defaultRole) {
-		Env.getCtx().remove(ROLE_KEY);
-		Env.getCtx().put(ROLE_KEY, defaultRole);
+		s_defaultrole.put(ROLE_KEY, defaultRole);
```

Comment:
hi @CarlosRuiz-globalqss , this is not correct, that make it the default role for everyone (i.e global), not just the current user session. Like I've suggested earlier, you can check the existence of role in cache to determine whether you need to reload it but you can't directly get/put from it.

Reply:
Thanks @hengsin - implemented your suggestion, tested and it works perfect.